config/base.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/notaryproject/notation-go/dir"
+)
+
+var (
+	// ConfigPath is the path for config.json
+	ConfigPath string
+	// SigningKeysPath is the path for signingkeys.json
+	SigningKeysPath string
+
+	// fileInfo is the information of config.json
+	fileInfo     *File
+	fileInfoOnce sync.Once
+
+	// signingKeysInfo is the information of signingkeys.json
+	signingKeysInfo     *SigningKeys
+	signingKeysInfoOnce sync.Once
+)
+
+func init() {
+	ConfigPath = dir.Path.Config()
+	SigningKeysPath = dir.Path.SigningKeyConfig()
+}
+
+// Configuration is an interface to manage notation config
+type Configuration interface {
+	Save() error
+}
+
+// Config is the main config struct of notation-go
+type Config struct {
+	*File
+	*SigningKeys
+}
+
+// Save stores sub-configurations to files
+func (c *Config) Save() error {
+	if err := c.File.Save(); err != nil {
+		return err
+	}
+	if err := c.SigningKeys.Save(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// LoadOnce returns the previously read config file.
+// If previous config file does not exist, it reads the config from file
+// or return a default config if not found.
+// The returned config is only suitable for read only scenarios for short-lived processes.
+func LoadOnce() (*Config, error) {
+	fileInfo, err := loadFileOnce()
+	if err != nil {
+		return nil, err
+	}
+	signingKeysInfo, err := loadSigningKeysOnce()
+	if err != nil {
+		return nil, err
+	}
+	return &Config{
+		File:        fileInfo,
+		SigningKeys: signingKeysInfo,
+	}, nil
+}
+
+// Save stores the config to file
+func Save(filePath string, config interface{}) error {
```

Comment:
The second argument "config interface{}" here actually could be either config.json or signingkeys.json, rename this to ''file interface{}" might be clearer.

Reply:
The `config interface{}` is a pointer of a struct, so named it config is better?

Reply:
Changed `config` to be `cfg`